andela/ah-robotics-frontend
# #162949039 Implement User Profile

#### What does this PR do?
Enable user to view and edit their profile
#### Description of Task to be completed?
#### How should this be manually tested?
1. Navigate to a folder where you would prefer to test the repository.

2. Create a directory
 ```$```  ```mkdir myproject```

3. Navigate to the directory "myproject"
 ```$```   ```cd myproject```
4. Clone the author's haven repository
  ```$```   ```git clone https://github.com/andela/ah-robotics-frontend.git```
5. Change the directory into the "ah-robotics-fronted"
  ```$```   ```cd ah-robotics-frontend```
6.  Checkout to the branch ft-user-profile-page-162949039
 ```$```   ```git checkout ft-user-profile-page-162949039```
7. Install dependancies from the package.json
 ```$```   ```npm install``` to install all the modules listed in the `package.json` file.

8. Start the application in development mode
  ```$```   ```npm start``` to run the app in the development mode 

The application opens http://localhost:3000 in the browser.
Change the URL on the browser to;
 ```
https://ah-robotics-frontend-sta-pr-13.herokuapp.com

```

The browser will show like represented in screenshot _**image: 1**_

9. Click on profile on the Navbar and enter a valid email **for demo purpose use**;

**Email:** ```steven.otieno1@gmail.com``` 
**Password:** ```test123#```

and click on the ```login``` button, see _**image: 2**_.

10. On the welcome page, as in  _**Image: 3**_, click on ```profile``` on the navbar.
The profile is displayed as shown in _**image 4**_

11. Click on the ```Edit Profile``` **button** it will lead you to _**image 5**_ to make changes.

In the Bio input text area, type in what you want to have as your **bio**, e.g qualifications, interest and the **Click** on the ```Save``` **button** once you have completed.

See _**image: 6a**_ and _**image: 6b**_ before and after click on ```Save``` **button**.

12. **Nb:** The ```Cancel``` **button** will return you to the profile page if you choose not to save changes. 

13. To run tests, open the terminal and run the following command:
See **_Image: 7_**
```
npm test
``` 
14. To run tests and coverage, open the terminal and run the following command:
```
npm test -- --coverage
```
<img width="871" alt="screenshot 2019-02-21 at 07 14 45" src="https://user-images.githubusercontent.com/39020475/53143329-8c87d200-35a8-11e9-9510-2ab196231fd2.png">

#### Any background context you want to provide?
#### What are the relevant pivotal tracker stories?
[#162949039](https://www.pivotaltracker.com/story/show/162949039)
#### Screenshots (if appropriate)
##### Image:1 Authors Haven landing page

<img width="1234" alt="screenshot 2019-02-27 at 10 45 16" src="https://user-images.githubusercontent.com/39020475/53473964-ebec5300-3a7c-11e9-803b-bdc5fc13cbda.png">


##### Image: 2 Login page
<img width="1248" alt="screenshot 2019-02-22 at 07 35 08" src="https://user-images.githubusercontent.com/39020475/53220259-a0990580-3674-11e9-9329-b060b9a4f788.png">

##### Image: 3 Welcome page _the username_  for steven.otieno1@gmail.com is _aspine_

<img width="1234" alt="screenshot 2019-02-27 at 10 48 50" src="https://user-images.githubusercontent.com/39020475/53474135-5a311580-3a7d-11e9-9aaa-69852ab07532.png">

##### Image: 4 Profile page 

<img width="1234" alt="screenshot 2019-02-27 at 10 51 56" src="https://user-images.githubusercontent.com/39020475/53474312-cc095f00-3a7d-11e9-8f99-4f0058007c52.png">

##### Image: 5 Profile Edit Page.

<img width="1248" alt="screenshot 2019-02-22 at 07 56 19" src="https://user-images.githubusercontent.com/39020475/53220819-89a7e280-3677-11e9-9d46-9d362ceb3369.png">

##### Image: 6 a & b Editing profile sample.
**Image: 6a**

<img width="1234" alt="screenshot 2019-02-27 at 11 01 32" src="https://user-images.githubusercontent.com/39020475/53474792-222ad200-3a7f-11e9-9e32-f35a0e303bc7.png">

**image: 6b**
<img width="1234" alt="screenshot 2019-02-27 at 10 54 00" src="https://user-images.githubusercontent.com/39020475/53474545-69fd2980-3a7e-11e9-880c-30532c64584a.png">



##### Image: 7 Test run on branch
<img width="860" alt="screenshot 2019-02-21 at 07 23 11" src="https://user-images.githubusercontent.com/39020475/53143581-a37af400-35a9-11e9-833c-4204e0795ace.png">
#### Questions:
